docs/DeveloperGuide.adoc

```diff
@@ -154,11 +154,137 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 This section describes some noteworthy details on how certain features are implemented.
 
 // tag::undoredo[]
+=== Import feature
+==== Implementation
+
+The import feature allows users to import data from a comma-separated values (CSV) file.
+It allows users to bulk import their contacts from a previously exported data from CoderLifeInsights application.
+
+Given below is the sequence diagram shows how the import operation works with the command `import life.csv`:
+
+.Interaction for `import` Command operation between Logic Component and Model Component.
+[#ImportFileSequenceDiagram,align="center"]
+image::ImportFileSequenceDiagram.png[]
+
+The following is a brief explanation of <<ImportFileSequenceDiagram>> :
+
+1. User enters `import life.csv`.
+2. `AddressBookParser#parseCommand()` is being called by `LogicManager`.
+3. This would create `ImportCommand` and checks if the input is valid.
+4. Feedback would be returned to `AddressBookParser` and then to `LogicManager`.
+5. `LogicManager` would then call `execute()` and creates `ImportFile`.
+6. `ImportFile` would then call `importCsv()` to return a `List<Person>`.
+7. `Model#importCsvToAddressBook(importedPeople)` is then being called by `LogicManager`
+passing on `List<Person>` to call `VersionedAddressBook#addPersons(importedPeople)`
+8. `VersionedAddressBook` would call the method `addAll` and populate the data accordingly.
+9. On successful import with no duplicates of person, it would return a success message to `LogicManager`
+
+The following is an activity diagram when a user executes the import command:
+
+.Import feature activity diagram.
+[#ImportFileActivityDiagram,align="center"]
+image::ImportFileActivityDiagram.png[]
+
+When a user executes the `import` command, a parser would check if the CSV is valid or not.
```

Comment:
is the csv file stored within the application or somewhere else in the system? (e.g. OS) if its the latter, will the parser check to see whether the file exists within a directory to determine whether it is valid?